ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/repository/FragmentRepository.java

```diff
@@ -58,4 +58,13 @@ Optional<Fragment> retrieveMutableFragment(String viewName,
 	Stream<Fragment> getDeletionCandidates();
 
 	void removeRelationsPointingToFragmentAndDeleteFragment(LdesFragmentIdentifier readyForDeletionFragment);
+
+	/**
+	 * Finds and updates all the child relations of the fragment and makes them immutable.
+	 * For example fragment year=2024, all fragments referring to months, days, hours, etc. within 2024 will become immutable.
+	 * The fragment itself does NOT become immutable. Only the children
+	 * When the fragment has no children, or only has immutable children, then no updates are done.
+	 * @param fragment The fragment whose children should become immutable
+	 */
+    void makeChildrenImmutable(Fragment fragment);
```

Comment:
tabbing seems off.. ?

Reply:
This must be a github thing. In intellij, this seems fine.